util/reggen/ip_block.py

```diff
@@ -348,3 +348,21 @@ def get_signal_by_name_as_dict(self, name: str) -> Dict[str, object]:
         else:
             raise ValueError("Signal {} does not exist in IP block {}"
                              .format(name, self.name))
+
+    def is_reg_shadowed(self) -> bool:
+        '''Return boolean indication whether reg block contains shadowed reigsters'''
```

Comment:
Could this be called `has_shadowed_reg` or something similar? Otherwise the name sounds like "is this register shadowed?", which isn't quite what you mean. Also (nit) the last word of the docstring is garbled: should be "re**gi**sters".

Reply:
yeah `has_shadowed` is better, let me change.